gmso/core/views.py

```diff
@@ -0,0 +1,143 @@
+from collections import defaultdict
+
+from gmso.core.angle import Angle
+from gmso.core.angle_type import AngleType
+from gmso.core.atom import Atom
+from gmso.core.atom_type import AtomType
+from gmso.core.bond import Bond
+from gmso.core.bond_type import BondType
+from gmso.core.dihedral import Dihedral
+from gmso.core.dihedral_type import DihedralType
+from gmso.core.improper import Improper
+from gmso.core.improper_type import ImproperType
+
+potential_attribute_map = {
+    Atom: "atom_type",
+    Bond: "bond_type",
+    Angle: "angle_type",
+    Dihedral: "dihedral_type",
+    Improper: "improper_type",
+}
+
+
+def get_identifier(potential):
+    """Get identifier for a topology potential."""
+    if isinstance(potential, AtomType):
+        return potential.name
+    if isinstance(potential, (BondType, AngleType, DihedralType, ImproperType)):
+        return potential.member_types or potential.member_classes
+
+
+def unique_potentials(potential_types):
+    """Filter unique potentials based on pre-defiend identifiers."""
+    visited = defaultdict(set)
+    for potential_type in potential_types:
+        identifier = get_identifier(potential_type)
+        if identifier not in visited[type(potential_type)]:
+            visited[type(potential_type)].add(identifier)
+
+            yield potential_type
+
+
+class PotentialFilters:
+    UNIQUE_NAME_CLASS = "unique_name_class"
```

Comment:
Can we make other potential filters called "UNIQUE_EXPRESSION_CLASS" and "UNIQUE_PARAMETER_CLASS"?

```python
def get_expression(potential):
    """Get sympy expression for a topology potential."""

    return potential.expression

def unique_potentials_by_expression(potential_types):
    """Filter unique potentials based on pre-defined sympy expressions."""
    visited = defaultdict(set)
    for potential_type in potential_types:
        identifier = get_expression(potential_type)
        if identifier not in visited[type(potential_type)]:
            visited[type(potential_type)].add(identifier)

            yield potential_type

def get_parameters(potential):
    """Get sympy expression for a topology potential."""

    return (
        tuple(potential.get_parameters().keys()), 
        tuple(map(lambda x: x.to_value(), my_ex.get_parameters().values()))
    )

def unique_potentials_by_parameters(potential_types):
    """Filter unique potentials based on pre-defined sympy expressions."""
    visited = defaultdict(set)
    for potential_type in potential_types:
        identifier = get_parameters(potential_type)
        if identifier not in visited[type(potential_type)]:
            visited[type(potential_type)].add(identifier)

            yield potential_type
```

Reply:
Yeah, having a good filteration system is crucial. I can add these methods, with a more sophisticated way of handling filters.

Reply:
Added this